core/src/main/scala/kafka/security/authorizer/AclAuthorizer.scala

```diff
@@ -453,10 +452,6 @@ class AclAuthorizer extends Authorizer with Logging {
     }
   }
 
-  // Returns Map instead of SortedMap since most callers don't care about ordering. In Scala 2.13, mapping from SortedMap
-  // to Map is restricted by default
-  private def unorderedAcls: Map[Resource, VersionedAcls] = aclCache
```

Comment:
I don't remember why this was added (it came from the original SimpleAclAuthorizer). Do we not need it any more?

Reply:
It only seems to be used by the acls method here https://github.com/apache/kafka/pull/7882/files#diff-852b9cb2ceb2b85ec25b422f72c42620L232, so it doesn't seem very useful to me any more, as that method can just use aclCache directly. I tried compiling with Scala 2.13 and it worked.

Reply:
If we don't need to map from a SortedMap to a Map, then this method is not useful.

Reply:
In AclAuthorizer, we are not mapping SortedMap to a Map, so it is not needed.